central/graphql/resolvers/nodes.go

```diff
@@ -543,24 +518,23 @@ func (resolver *nodeResolver) PlottedNodeVulnerabilities(ctx context.Context, ar
 
 func (resolver *nodeResolver) Scan(ctx context.Context) (*nodeScanResolver, error) {
 	res, err := resolver.root.wrapNodeScan(resolver.data.GetScan(), true, nil)
-	if resolver.ctx == nil {
-		resolver.ctx = ctx
-	}
 	if err != nil || res == nil {
 		return nil, err
 	}
-	res.ctx = resolver.nodeScopeContext()
+	res.ctx = resolver.nodeScopeContext(ctx)
 	return res, nil
 }
 
 func (resolver *nodeResolver) UnusedVarSink(_ context.Context, _ RawQuery) *int32 {
 	return nil
 }
 
-func (resolver *nodeResolver) nodeScopeContext() context.Context {
+func (resolver *nodeResolver) nodeScopeContext(ctx context.Context) context.Context {
```

Comment:
Context of incoming requests is not expected to be nil. Therefore, for this and other scope context builders, check `ctx` is not nil and panic on dev build. That way we can catch if incorrect context is passed around. Remember that  `resolver.ctx` is set to `ctx` from request.

```
if ctx == nil {
  utils.Should("argument 'ctx' is nil")
}
```